File: tests/c#/10/TestUtilities.cs
namespace Hardstuck.GuildWars2.BuildCodes.V2.Tests;

static class TestUtilities {
	public static Dictionary<string, string> CodesInvalid  = new();
	public static Dictionary<string, string> CodesV1       = new();
	public static Dictionary<string, string> CodesV2       = new();
	public static Dictionary<string, string> CodesIngame   = new();
	public static Dictionary<string, string> CodesV2Binary = new();

	static TestUtilities() {
		Dictionary<string, string> dict = null!;
		string currentKey = null!;
		string currentAccumulator = string.Empty;
		foreach(var line_ in File.ReadLines("codes.ini"))
		{
			var comment = line_.IndexOf(';');
			var line = (comment > -1 ? line_[..comment] : line_).Trim();
			if(string.IsNullOrWhiteSpace(line)) continue;

			if(line.StartsWith('[') && line.EndsWith(']'))
			{
				dict = line[1..^1] switch {
					"Invalid"  => CodesInvalid,
					"V1"       => CodesV1,
					"V2"       => CodesV2,
					"Ingame"   => CodesIngame,
					"V2Binary" => CodesV2Binary,
				};
			}
			else
			{
				if(dict != CodesV2Binary)
				{
					var split = line.IndexOf('=');
					dict[line[..split].Trim()] = line[(split + 1)..].Trim();
				}
				else if(line == "<end>")
				{
					dict[currentKey] = currentAccumulator;
					currentAccumulator = string.Empty;
				}
				else
				{
					var split = line.IndexOf('=');
					if(split != -1)
					{
						currentKey = line[..split].Trim();
					}
					else
					{
						currentAccumulator += line;
					}
				}
			}
		}
	}
}

File: tests/php/8.1/TestUtilities.php
<?php namespace Hardstuck\GuildWars2\BuildCodes\V2\Tests;

class TestUtilities {
	/** @var array<string, string> */
	public static array $CodesInvalid  = [];
	/** @var array<string, string> */
	public static array $CodesV1       = [];
	/** @var array<string, string> */
	public static array $CodesV2       = [];
	/** @var array<string, string> */
	public static array $CodesIngame   = [];
	/** @var array<string, string> */
	public static array $CodesV2Binary = [];

	static function __construct_static() {
		$dict = null;
		$currentDict = null;
		$currentKey = null;
		$currentAccumulator = '';
		foreach(file(__DIR__."/../../common/codes.ini") as $line_)
		{
			$comment = strpos($line_, ';');
			$line = trim($comment !== false ? substr($line_, 0, $comment) : $line_);
			if(empty($line)) continue;

			if(str_starts_with($line, '[') && str_ends_with($line, ']'))
			{
				$currentDict = substr($line, 1, strlen($line) - 2);
				switch ($currentDict) {
					case "Invalid" : $dict = &TestUtilities::$CodesInvalid; break;
					case "V1"      : $dict = &TestUtilities::$CodesV1; break;
					case "V2"      : $dict = &TestUtilities::$CodesV2; break;
					case "Ingame"  : $dict = &TestUtilities::$CodesIngame; break;
					case "V2Binary": $dict = &TestUtilities::$CodesV2Binary; break;
				};
			}
			else
			{
				if($currentDict !== 'V2Binary')
				{
					list($key, $value) = explode('=', $line, 2);
					$dict[trim($key)] = trim($value);
				}
				else if($line === "<end>")
				{
					$dict[$currentKey] = $currentAccumulator;
					$currentAccumulator = '';
				}
				else
				{
					$split = strpos($line, '=');
					if($split !== false)
					{
						$currentKey = trim(substr($line, 0, $split));
					}
					else
					{
						$currentAccumulator .= $line;
					}
				}
			}
		}
	}
}

TestUtilities::__construct_static();